### PR TITLE
feat: added refresh recommend question api

### PR DESCRIPTION
**Pull request by Icemap**
Added an API that can refresh the recommended questions

### CODE REVIEW

**Review by IANTHEREAL — APPROVED** [review]

File: backend/app/api/routes/chat.py [hunk 1]
```diff
@@ -229,3 +229,14 @@ def get_recommend_questions(session: SessionDep, chat_message_id: int):
 
     return get_chat_message_recommend_questions(session, chat_message)
 
+
+@router.post("/chat-messages/{chat_message_id}/recommend-questions", response_model=List[str])
+def refresh_recommend_questions(session: SessionDep, chat_message_id: int):
```
Comment (sykp241095) [inline backend/app/api/routes/chat.py:234]: How about using an adj "recommended" ?

Comment (Icemap) [reply backend/app/api/routes/chat.py]: @634750802 The front-end needs to change it as well.

File: backend/app/api/routes/chat.py [hunk 2]
```diff
@@ -229,3 +229,14 @@ def get_recommend_questions(session: SessionDep, chat_message_id: int):
...
+
+@router.post("/chat-messages/{chat_message_id}/recommended-questions", response_model=List[str])
+def refresh_recommended_questions(session: SessionDep, chat_message_id: int):
+    chat_message = chat_repo.get_message(session, chat_message_id)
+    if not chat_message or len(chat_message.content) == 0:
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND, detail="Chat message not found"
+        )
+
+    remove_chat_message_recommend_questions(session, chat_message_id)
```
Comment (634750802) [inline backend/app/api/routes/chat.py:241]: This will cause empty recommended questions list when error occur at generating new questions

Comment (Icemap) [reply backend/app/api/routes/chat.py:241]: The next time call we can leverage the get function to generate it again.

**Pull request merged**